src/components/Attachment/Attachment.tsx

```diff
@@ -162,7 +163,7 @@ class Attachment extends UIComponent<Extendable<AttachmentProps>, any> {
           </div>
         )}
         {!_.isNil(progress) &&
-          createHTMLDivision('', {
+          createSlotFactory('div')(progress, {
```

Comment:
I would not expect every call of this to create a new factory.  Seems the factory should internally handle mapping the appropriate mapValToProps function to the element type instead of creating new factory functions on every invocation.